sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/ConfiguringLoggersTest.cs

```diff
@@ -0,0 +1,157 @@
+                new SonarAnalyzer.Rules.VisualBasic.ConfiguringLoggers(new TestAnalyzerConfiguration(null, "S4792")),
+                additionalReferences: NLogReferences);
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        [TestCategory("Hotspot")]
+        public void ConfiguringLoggers_Serilog_CS()
+        {
+            Verifier.VerifyAnalyzer(@"TestCases\ConfiguringLoggers_Serilog.cs",
+                new ConfiguringLoggers(new TestAnalyzerConfiguration(null, "S4792")),
+                additionalReferences: SeriLogReferences);
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        [TestCategory("Hotspot")]
+        public void ConfiguringLoggers_Serilog_VB()
+        {
+            Verifier.VerifyAnalyzer(@"TestCases\ConfiguringLoggers_Serilog.vb",
+                new SonarAnalyzer.Rules.VisualBasic.ConfiguringLoggers(new TestAnalyzerConfiguration(null, "S4792")),
+                additionalReferences: SeriLogReferences);
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        [TestCategory("Hotspot")]
+        public void ConfiguringLoggers_CS_RuleDisabled()
+        {
+            Verifier.VerifyNoIssueReported(@"TestCases\ConfiguringLoggers_AspNetCore.cs",
+                new ConfiguringLoggers(),
+                additionalReferences: AspNetCoreLoggingReferences);
+        }
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        [TestCategory("Hotspot")]
+        public void ConfiguringLoggers_VB_RuleDisabled()
+        {
+            Verifier.VerifyNoIssueReported(@"TestCases\ConfiguringLoggers_AspNetCore.vb",
+                new SonarAnalyzer.Rules.VisualBasic.ConfiguringLoggers(),
+                additionalReferences: AspNetCoreLoggingReferences);
+        }
+        
+        private static IEnumerable<MetadataReference> AspNetCoreLoggingReferences =>   
+            FrameworkMetadataReference.Netstandard
+            .Concat(NuGetMetadataReference.MicrosoftAspNetCore(Constants.NuGetLatestVersion))
+            .Concat(NuGetMetadataReference.MicrosoftAspNetCoreHosting(Constants.NuGetLatestVersion))
+            .Concat(NuGetMetadataReference.MicrosoftAspNetCoreHostingAbstractions(Constants.NuGetLatestVersion))
+            .Concat(NuGetMetadataReference.MicrosoftAspNetCoreHttpAbstractions(Constants.NuGetLatestVersion))
+            .Concat(NuGetMetadataReference.MicrosoftExtensionsConfigurationAbstractions(Constants.NuGetLatestVersion))
+            .Concat(NuGetMetadataReference.MicrosoftExtensionsDependencyInjectionAbstractions(Constants.NuGetLatestVersion))
+            .Concat(NuGetMetadataReference.MicrosoftExtensionsOptions(Constants.NuGetLatestVersion))
+            .Concat(NuGetMetadataReference.MicrosoftExtensionsLoggingPackages(Constants.NuGetLatestVersion))
+        ;
+
+        private static IEnumerable<MetadataReference> Log4NetReferences =>
+            NuGetMetadataReference.Log4Net(Constants.NuGetLatestVersion, "net45-full")
```

Comment:
I'm explicitly specifying which set of assemblies within the NuGet package to use (those in the "net45-full" folder) because the default target framework selection logic in _NuGetMetadataReference_ picks a version that leads to compile errors.